src/cpu/jit_uni_eltwise.cpp

```diff
@@ -724,108 +724,231 @@ template struct jit_uni_eltwise_injector_f32<sse41>;
 
 
 struct jit_args {
-    const float *from;
-    const float *for_comparison;
-    const float *to;
+    const void *from;
+    const void *for_comparison;
+    const void *to;
     size_t work_amount;
 };
 
-struct jit_uni_eltwise_kernel_f32 : public c_compatible {
+struct jit_uni_eltwise_kernel : public c_compatible {
+    enum {
+        _rnd_trunc = 3u // Round toward zero
+    };
+
     const eltwise_desc_t &desc_;
 
     void (*ker_)(const jit_args *);
-    void operator()(const jit_args *args) { assert(ker_); ker_(args); }
+    void operator()(const jit_args *args) {
+        assert(ker_);
+        ker_(args);
+    }
 
-    jit_uni_eltwise_kernel_f32(const eltwise_desc_t &desc)
+    jit_uni_eltwise_kernel(const eltwise_desc_t &desc)
         : desc_(desc), ker_(nullptr) {}
-    virtual ~jit_uni_eltwise_kernel_f32() {}
+    virtual ~jit_uni_eltwise_kernel() {}
 
 protected:
     bool is_bwd() const { return desc_.prop_kind == prop_kind::backward_data; }
+    mkldnn_data_type_t data_type() const { return desc_.data_desc.data_type; }
+    size_t data_type_size() const { return types::data_type_size(data_type()); }
 };
 
 /* jit kernels */
 namespace {
 
 template <cpu_isa_t isa>
-struct jit_uni_relu_kernel_f32 : public jit_uni_eltwise_kernel_f32,
+struct jit_uni_relu_kernel : public jit_uni_eltwise_kernel,
     public jit_generator
 {
-    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_uni_relu_kernel_f32)
-
-    void compute_step(bool vectorize, const int uf, const int shift) {
-        for (int i = 0; i < uf; i++) {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_uni_relu_kernel)
+
+    void compute_step(bool vectorize, const size_t uf, const size_t shift) {
```

Comment:
The nested `if` statements make the code difficult to follow. Does it make sense to separate into functions, named something like `compute_step_float` and `compute_step_int` ?

Reply:
@shelleygoel 
I've made refactoring on my own and reduce number of `if` in one function.
Take a look at my commit [b0f13c8](https://github.com/intel/mkl-dnn/pull/443/commits/b0f13c8a9dbbf5fa1faa182bf14e0a6ab2a0202b)

Reply:
Hi @basargin Sorry for the delayed response :( Please see my review.